internal/uvm/network.go

```diff
@@ -119,12 +121,16 @@ func GetNamespaceEndpoints(ctx context.Context, netNS string) ([]*hns.HNSEndpoin
 
 // NCProxyEnabled returns if there is a network configuration client.
 func (uvm *UtilityVM) NCProxyEnabled() bool {
-	return uvm.ncProxyClient != nil
+	return uvm.ncProxyClientAddress != ""
 }
 
-// NCProxyClient returns the network configuration proxy client.
-func (uvm *UtilityVM) NCProxyClient() ncproxyttrpc.NetworkConfigProxyService {
-	return uvm.ncProxyClient
+func (uvm *UtilityVM) GetNCProxyClient() (ncproxyttrpc.NetworkConfigProxyService, error) {
```

Comment:
`Get` as a verb in the name doesn't really make it clear what the impact of calling this function is (we do a completely new socket dial). I would suggest changing the name to something that makes it clear that we're not just returning some cached connection.